src/test/scala/io/citrine/lolo/bags/CorrelationStudy.scala

```diff
@@ -0,0 +1,441 @@
+          runTrial(
+            function = function,
+            numTrain = numTrain,
+            numTest = numTest,
+            numBags = numTrain,
+            rhoTrain = rhoTrain,
+            quadraticCorrelationFuzz = fuzz,
+            samplingNoise = samplingNoise,
+            observational = observational,
+            rng = thisRng
+          )
+        }
+      case Bags =>
+        numBags => Iterator.tabulate(numTrials) { _ =>
+          val thisRng = new Random(rng.nextLong())
+          runTrial(
+            function = function,
+            numTrain = numTrain,
+            numTest = numTest,
+            numBags = numBags.toInt,
+            rhoTrain = rhoTrain,
+            quadraticCorrelationFuzz = quadraticCorrelationFuzz,
+            samplingNoise = samplingNoise,
+            observational = observational,
+            rng = thisRng
+          )
+        }
+      case NumTraining(optionNumBags) =>
+        numTraining => Iterator.tabulate(numTrials) { _ =>
+          val thisRng = new Random(rng.nextLong())
+          runTrial(
+            function = function,
+            numTrain = numTraining.toInt,
+            numTest = numTest,
+            numBags = optionNumBags.getOrElse(numTraining.toInt),
+            rhoTrain = rhoTrain,
+            quadraticCorrelationFuzz = quadraticCorrelationFuzz,
+            samplingNoise = samplingNoise,
+            observational = observational,
+            rng = thisRng
+          )
+        }
+    }
+    Merit.plotMeritScan(
+      parameterName = variedParameter.name,
+      parameterValues = parameterValues,
+      merits = merits,
+      rng = new Random(0L) // this is irrelevant unless we were computing Uncertainty Correlation
+    )(pvaBuilder)
+  }
+
+  def runTrial(
+                function: TrueFunction,
+                numTrain: Int,
+                numTest: Int,
+                numBags: Int,
+                rhoTrain: Double,
+                quadraticCorrelationFuzz: Double,
+                samplingNoise: Double,
+                observational: Boolean,
+                rng: Random
+              ): (PredictionResult[Seq[Any]], Seq[Seq[Any]]) = {
+    val dataGenSeed = rng.nextLong()
+    val dataNoiseSeed = new Random(rng.nextLong())
+    val trainRng = new Random(rng.nextLong())
+    val bagSeed = rng.nextLong()
+    val linearCorrelationRng = new Random(rng.nextLong())
+    val quadraticCorrelationRng = new Random(rng.nextLong())
+
+    val fullData: Seq[(Vector[Double], Double)] = TestUtils.generateTrainingData(
+      numTrain + numTest,
+      function.numCols,
+      noise = 0.0, // Add noise later, after computing covariate labels
+      function = function.function,
+      seed = dataGenSeed
+    )
+
+    val inputs: Seq[Vector[Double]] = fullData.map(_._1)
+    val realLabel: Seq[Double] = fullData.map(_._2)
+    val linearLabel: Seq[Double] = makeLinearCorrelatedData(realLabel, rhoTrain, linearCorrelationRng)
```

Comment:
Would it be helpful to try an example with a sinusoidal map on the test function output?

I do not have any prior that this would be useful, besides the fact that it's a natural extension to a different function type.

Reply:
I think that the quadratic function covers a similar case, in that it has no global correlation between the variables but strong local correlation.

That said, I'm open to suggestions if you think this or other test functions are meaningfully different.